### PR TITLE
Manage accounts

### PR DESCRIPTION
**Pull request by annarog**
Through Manage Users tab: 
- admins are able to view all users (can search by role)
- admins can delete users 
- admins are able to register new users, specifying if they're "user" or "admin"

Through View Participants:
- all users can view the social workers assigned to each participant
- admins can edit social workers assigned (add/remove)

Also: fixed bug with OrderBy pipe where lower case and upper case letters compared differently (for example, "b" would appear after "W" in ascending order)

#63

### CODE REVIEW

**Comment by annarog** [comment]
Backend tests
screen shot 2018-03-02 at 7 38 54 pm

Frontend tests
screen shot 2018-03-02 at 7 39 10 pm

File: angular-client/src/app/components/users/register-user/register-user.component.ts [hunk 1]
```diff
@@ -9,7 +10,14 @@ import { Router } from '@angular/router';
...
-  user: any = {};
+  user: User = {
+    name: '',
+    pronouns: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+    role: 'user' // selected role is 'user' by default
+  };
   loading: boolean = false;
```
Comment (Ldhatter) [inline angular-client/src/app/components/users/register-user/register-user.component.ts]: My linter is crying, but sometimes I don't fully trust the default settings in this linter. In this case though, don't you think we could simply write it as `loading = false;`  (same for the other 2)?


screen shot 2018-03-03 at 10 27 11 pm

File: angular-client/src/app/components/modals/edit-worker-modal/edit-worker-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<h1 mat-dialog-title>Edit Assigned Social Workers</h1>
```
Comment (Ldhatter) [inline angular-client/src/app/components/modals/edit-worker-modal/edit-worker-modal.component.html]: @annarog  Where is this? when i run it, i can't find where this modal gets used.

File: angular-client/src/app/components/users/register-user/register-user.component.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<div fxLayout="row wrap" fxLayoutAlign="space-around center" fxFlex="100">
```
Comment (Ldhatter) [inline angular-client/src/app/components/users/register-user/register-user.component.html]: We should convert this reactive. But we can make another PR to refactor this.

File: angular-client/src/app/components/users/register-user/register-user.component.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  <mat-card fxFlex="50">
+    <mat-card-title>
+      Register a User
+    </mat-card-title>
+    <form id="registrationForm" name="registrationForm" (ngSubmit)="signUp()" *ngIf="!loading">
+      <mat-card-content>
+
+        <div class="form-group">
```
Comment (Ldhatter) [inline angular-client/src/app/components/users/register-user/register-user.component.html]: if you want to style a bunch of mat-form-fields, there's no need to wrap it with a div. you can just add in the css `mat-form-field { color: something....}`

so these divs could be removed :)

File: angular-client/src/app/components/users/register-user/register-user.component.ts (hunk 1 above)
Comment (annarog) [reply angular-client/src/app/components/users/register-user/register-user.component.ts]: Yea no problem :)

File: angular-client/src/app/components/modals/edit-worker-modal/edit-worker-modal.component.html (hunk 2 above)
Comment (annarog) [reply angular-client/src/app/components/modals/edit-worker-modal/edit-worker-modal.component.html]: You need to be signed in as Admin, only admins can do this. See the description of this PR for what else they can do :)

File: angular-client/src/app/components/users/register-user/register-user.component.html (hunk 4 above)
Comment (annarog) [reply angular-client/src/app/components/users/register-user/register-user.component.html]: Yea will do. This actually comes from old code so I didn't notice

File: angular-client/src/app/components/modals/edit-worker-modal/edit-worker-modal.component.html (hunk 2 above)
Comment (Ldhatter) [reply angular-client/src/app/components/modals/edit-worker-modal/edit-worker-modal.component.html]: ok found it! wow neato

File: angular-client/src/app/components/users/register-user/register-user.component.ts (hunk 1 above)
Comment (Ldhatter) [reply angular-client/src/app/components/users/register-user/register-user.component.ts]: fixed in 7a09003

**Comment by Ldhatter** [comment]
@adriannadiaz  can you confirm changes and close this?

**Comment by annarog** [comment]
@adriannadiaz please review the changes :) As for the Edit Social Worker functionality I propose to do it as part of one of the tickets for next iteration
